packages/babel-plugin-jsx-dom-expressions/src/dom/template.js

```diff
@@ -129,65 +129,55 @@ function wrapDynamics(path, dynamics) {
       ])
     );
   }
-  const declarations = [],
-    statements = [],
-    identifiers = [],
-    prevId = t.identifier("_p$");
+
+  const statements = [];
+
   dynamics.forEach(({ elem, key, value, isSVG, isCE, tagName }) => {
     const identifier = path.scope.generateUidIdentifier("v$");
+    path.scope.push({ kind: "let", id: identifier });
```

Comment:
There's one issue with this, it adds these dirtychecking variables via `path.scope.push`, which means that the variables are placed *outside* of the template IIFE.

`results.declarations` is used to add constant variables for traversing the elements, but they're hardwired for that, and I'm not exactly sure how deep the rabbit hole goes for adding a new `dyn_declarations` array that's just meant to hold these variables.

Perhaps the ideal approach here would probably be to replace itself with an IIFE when entering a JSX element node from a foreign node, I think this would allow Babel to understand that it should probably place these variables *inside* the IIFE we just created, and we wouldn't have the need for `results.declarations` and `results.decl` at all.

Reply:
though I suppose one question remains and that's how exactly it should be dealing with the `template` calls that are supposed to be hoisted out of our IIFE, but that shouldn't be that big of an issue